src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1274,20 +1274,24 @@
     (fs/delete csr-path)
     (log/debug (i18n/trs "Removed certificate request for {0} at ''{1}''" subject csr-path))))
 
-(schema/defn revoke-existing-cert!
-  "Revoke the subject's certificate. Note this does not destroy the certificate.
-   The certificate will remain in the signed directory despite being revoked."
+(schema/defn revoke-existing-certs!
+  "Revoke the subjects' certificates. Note this does not destroy the certificates.
+   The certificates will remain in the signed directory despite being revoked."
   [{:keys [signeddir cacert cacrl cakey]} :- CaSettings
-   subject :- schema/Str]
-  (let [serial  (-> (path-to-cert signeddir subject)
-                    (utils/pem->cert)
-                    (utils/get-serial))
-        new-crl (utils/revoke (utils/pem->crl cacrl)
-                              (utils/pem->private-key cakey)
-                              (.getPublicKey (utils/pem->ca-cert cacert cakey))
-                              serial)]
-    (utils/crl->pem! new-crl cacrl)
-    (log/debug (i18n/trs "Revoked {0} certificate with serial {1}" subject serial))))
+   subjects :- [schema/Str]]
+  (let [original-crl (utils/pem->crl cacrl)
+        ca-private-key (utils/pem->private-key cakey)
+        ca-public-key (.getPublicKey (utils/pem->ca-cert cacert cakey))
+        serials (reduce (fn [serials subject]
+                          (conj serials (-> (path-to-cert signeddir subject)
```

Comment:
not a big deal, but this is effectively a map.

Reply:
Oh you're right, thanks. Trying to get more fluent with these functional transformers.